MetaMask/metamask-extension
# Wait for Snap accounts to be removed before removing the snap

## **Description**

Fixes: https://github.com/MetaMask/accounts-planning/issues/76

This PR updates the `KeyringController` to await account removal, and modifies the UI to wait until all accounts of a snap have been removed before removing the snap (this only affects Accounts Snaps).

The bug linked above was occurring because the snap would be removed before all accounts could be removed from the snap, due to the Keyring Controller missing an `await`.

## **Manual testing steps**

1. Install Flask and the SSK Snap.
2. Create some accounts (3 or more) using the SSK.
3. Remove the SSK Snap.
4. You should **not** see the following error in the logs:

    ![image](https://github.com/MetaMask/metamask-extension/assets/68558152/b0f6279e-f07e-42fc-9012-ca749c85a693)

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
